packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -395,10 +391,31 @@ export default class CodeMode extends Component<Signature> {
     return this.moduleContentsResource?.declarations || [];
   }
 
+  private get nextSelectedDeclaration() {
```

Comment:
what do you mean by "nextSelectedDeclaration", do you mean the declaration adjacent to the selected declaration? i'm a little confused what "next" means in this context

Reply:
When navigating from one module from another by clicking on a definition, the nextSelectedDeclaration really just represents the selection in the module navigated to. 

- you are at employee.gts, you click person.gts 
- person.gts opens as new module. The next selected declaration represents the `Person` in the list of declarations in the person file

For more clarity and less definition mangling, I can move the code in nextSelectedDeclaration into selectedDeclaration getter

Reply:
i get what you mean, but "next" is a very transitory state that it's in before the callback is executed. by the time the callback is executed and the state is set, it's not "next" anymore--it''s the current selected declaration. i think absorbing this into the selectedDeclaration will make it easier to understand

Reply:
Actually, there is really no notion of next. The thinking should be more like

"I open a module. If I have that code ref existing in my module, I select it."

```
 private get selectedDeclaration() {
    if (this._selectedDeclaration) {
      return this._selectedDeclaration;
    } else {
      if (this.nextSelectedDeclaration) {
        return this.nextSelectedDeclaration;
      }
      return this.declarations.length > 0 ? this.declarations[0] : undefined;
    
```

This might be a better naming

```
  private get selectedDeclaration() {
    if (this._userSelecteDeclaration) {
      // when user selects declarations by clicking
      return this._userSelectedDeclaration;
    } else {
      // when navigating from somewhere, check if code ref exists, if so, select corresponding declaration with code ref 
      if (this._selectedDeclaration) {
        return this._selectedDeclaration;
      }
      // default to 1st selection
      return this.declarations.length > 0 ? this.declarations[0] : undefined;
    }
  }
```

Reply:
Yea ur right. there is no notion of next

"I open a module. If I have that code ref existing in my module, I select it."

```
 private get selectedDeclaration() {
    if (this._selectedDeclaration) {
      return this._selectedDeclaration;
    } else {
      if (this.nextSelectedDeclaration) {
        return this.nextSelectedDeclaration;
      }
      return this.declarations.length > 0 ? this.declarations[0] : undefined;
    
```

This might be a better naming

```
  private get selectedDeclaration() {
    if (this._userSelecteDeclaration) {
      // when user selects declarations by clicking
      return this._userSelectedDeclaration;
    } else {
      // when navigating from somewhere, check if code ref exists, if so, select corresponding declaration with code ref 
      if (this._selectedDeclaration) {
        return this._selectedDeclaration;
      }
      // default to 1st selection
      return this.declarations.length > 0 ? this.declarations[0] : undefined;
    }
  }
```

wdy think?

Reply:
yeah that makes much more sense: :+1:

Reply:
> another option, if you want to keep the method separate still would be to turn it into a pure function and move it into module scope (so there is no `this` within it). that way it's clear that it's not state on the component. and you could name the function `resolveSelectedDeclaration()` or something similar....

ok I can see how this would make this clearer. To make this look more as a utility

Reply:
Makes lots of sense 👍